Client/src/Hooks/WdkServiceHook.ts

```diff
@@ -3,10 +3,12 @@ import { useState } from "react";
 import { useDispatch } from "react-redux";
 import { notifyUnhandledError } from "wdk-client/Actions/UnhandledErrorActions";
 
-interface WdkServiceCallback<T> {
-  (wdkService: WdkService): Promise<T>;
+export interface ServiceCallback<S, T> {
```

Comment:
Should `S extend WdkService`?

Reply:
Good question! I had intentionally left off that constraint because currently, a wrapped WdkService need not conform to the `WdkService` type. That said, I will go ahead and have `S extends WdkService`, because in practice, we probably shouldn't be working with wrapped `WdkService`s that don't extend `WdkService`.

(Possible topic of future work: when our bootstrap/initialization code is converted to TypeScript, require the return value of `wrapWdkService` to extend `WdkService`.)